Library/Homebrew/service.rb

```diff
@@ -355,6 +355,23 @@ def command
       @run&.map(&:to_s)
     end
 
+    sig { params(value: T.nilable(Integer)).returns(T.nilable(Integer)) }
+    def nice(value = nil)
+      case T.unsafe(value)
+      when nil
+        @nice
+      when Integer
+        if (T.must(value) < -20) || (T.must(value) > 19)
```

Comment:
Do you need `sudo` for negative `nice`?

Reply:
I guess a warning when you call `brew services` without `sudo` suffices here. But CC @SMillerDev and @MikeMcQuaid for thoughts here.

Reply:
Since this is not something that should happen I think throwing an error is the best course of action.

Reply:
It seems reasonable to truncate the `nice` level if you don't want to run `brew services` with `sudo` (which is what will happen automatically), though, so throwing an error seems a bit much here.

Reply:
I believe `requires_root` still allows you to run as a user, it'll just warn against doing so. So if we require that with the nice level you can still opt to have it truncated, brew will just suggest the fully functional option.

Reply:
Sounds like we should just have `nice < 0` imply `requires_root` then, regardless of whether the `service` block includes it. Not sure what the value of forcing its inclusion by throwing an error is.

Reply:
It would make it clear to developers and maintainers why it requires root. Instead of only telling users that they need to use root.

Reply:
I don't think an extra line that says `require_root true` is going to make the reason more transparent.